### PR TITLE
feat(go): add UpperBound interface for treemap

### PR DESCRIPTION
**Pull request by Kybxd**
refer: https://en.cppreference.com/w/cpp/container/map/upper_bound

### CODE REVIEW

File: pkg/treemap/treemap.go [hunk 1]
```diff
@@ -81,12 +81,26 @@ func (m *TreeMap[K, V]) Max() (key K, value V, ok bool) {
...
 	if found {
-		return TreeMapIterator[K, V]{m.tree.IteratorAt(node)}
+		iter = m.tree.IteratorAt(node)
 	}
+	return TreeMapIterator[K, V]{iter}
+}
+
+// UpperBound returns the iterator of the smallest key that is greater than the given key.
+// If key is not found, this returns the iterator past the last element (one-past-the-end).
+// This may either because the map is empty, or because all keys in the map are smaller than or equal to the given key.
```
Comment (wenchy) [inline pkg/treemap/treemap.go:95]: Improve clean comments:

>UpperBound returns an iterator pointing to the first element that is greater than key. If no such element is found, a past-the-end iterator is returned.

refer: https://en.cppreference.com/w/cpp/container/map/upper_bound

File: pkg/treemap/treemap.go [hunk 2]
```diff
@@ -81,12 +81,26 @@ func (m *TreeMap[K, V]) Max() (key K, value V, ok bool) {
 // If key is not found, this returns the iterator past the last element (one-past-the-end).
 // This may either because the map is empty, or because all keys in the map are smaller than the given key.
 func (m *TreeMap[K, V]) LowerBound(key K) TreeMapIterator[K, V] {
+	iter := m.tree.Iterator()
```
Comment (wenchy) [inline pkg/treemap/treemap.go:84]: Improve clean comments:

>LowerBound returns an iterator pointing to the first element that is not less than key. If no such element is found, a past-the-end iterator is returned.

refer: https://en.cppreference.com/w/cpp/container/map/lower_bound

**Review by wenchy — CHANGES_REQUESTED** [review]

File: pkg/treemap/treemap.go (hunk 1 above)
Comment (Kybxd) [reply pkg/treemap/treemap.go]: changed

File: pkg/treemap/treemap.go (hunk 2 above)
Comment (Kybxd) [reply pkg/treemap/treemap.go:83]: changed

**Review by wenchy — APPROVED** [review]

**Pull request merged**